pkg/instrumentation/database.go

```diff
@@ -2,159 +2,21 @@ package instrumentation
 
 import (
 	"context"
-	"fmt"
-	"strings"
 
-	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
-	ddtrace "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 	"gorm.io/gorm"
 )
 
-type spanContextKey string
-
-var (
-	// ParentSpanGormKey is the name of the parent span key
-	parentSpanGormKey = spanContextKey("trancingParentSpan")
-	// SpanGormKey is the name of the span key
-	spanGormKey = spanContextKey("tracingSpan")
-)
-
-// TraceDatabase sets span to gorm settings, returns cloned DB
+// TraceDatabase fetches the span from the context and injects it to a new
+// database session's statement context.
 func TraceDatabase(ctx context.Context, db *gorm.DB) *gorm.DB {
 	if ctx == nil {
 		return db
 	}
 
-	parentSpan, _ := ddtrace.SpanFromContext(ctx)
+	parentSpan, _ := tracer.SpanFromContext(ctx)
```

Comment:
[q]: Do we need the `TraceDatabase` function now? 🤔  We used to have it in order to propagate the span keys to context for instrumentation purposes. But now, since we use `dd-trace-go` it looks like we can remove `TraceDatabase` function completely ✂️

Reply:
Yes we do. for [database middleware](https://github.com/scribd/go-sdk/blob/main/pkg/middleware/database.go#L30) and interceptor, as we need a functionality to fetch the `tracer` from context and add it to the database.

Reply:
@laynax 

>for database middleware and interceptor, as we need a functionality to fetch the tracer from context and add it to the database.

That is my point, it looks like we don't need it anymore. We used to use it to [propagate the proper context](https://github.com/scribd/go-sdk/pull/106/files#diff-2b52422b8ec8cbd7b7316c2553325d0c7e8bea0b005139182447457e8c365ec3L93-L102), but now [dd-trace-go does it](https://github.com/DataDog/dd-trace-go/blob/79237af6569cdc90f176551075655a7007c7ddfd/contrib/gorm.io/gorm.v1/gorm.go#L151). Basically, setting the context should be sufficient now, as far as I understand

Reply:
Yeah but then the db span would be parentless and will be detached from HTTP request span.

Reply:
@laynax 

Ah, I see now, good point 👍  Then we need to pass the request context to the db context using the [WithContext](https://gorm.io/docs/context.html#Single-Session-Mode) function as described in the [gorm documentation](https://gorm.io/docs/context.html#Integration-with-Chi-Middleware) (the same way we can pass context from interceptors).

By the way, for consistency, we can move the instrumented gorm connection into this file, which will be aligned with what we currently have for other clients, wdyt?

Reply:
@laynax 

>I'd rather removing NewConnection instead, as @fotos proposed. thoughts?

In this case, I'd rather keep the `NewConnection` function and make it return the instrumented `gorm` instance always. This keeps the signature the same, it we wouldn't change the service's code to use new function

Reply:
Yeah that's what I meant in gibberish 😅

Reply:
>How would the new approach look like?

The new [go-chassis](https://github.com/scribd/go-chassis/pull/168/files#diff-53cd25fc908d5e63cc8675afbd4ff0910572a985a7cfec27f4460eb674f3b98eR90)

Reply:
@Neurostep The `TraceDatabase` with its test file is removed 🧹